### PR TITLE
[JENKINS-56121] Add option to (not) show raw yaml in console

### PR DESCRIPTION
**Pull request by joyarzun**
https://issues.jenkins-ci.org/browse/JENKINS-56121

Based on https://github.com/jenkinsci/kubernetes-plugin/pull/451 @fredg02

Since the raw yaml can add quite some noise to the console log, an
option was added to disable it. Default is ShowRawYaml=true, so the
default behavior does not change.

### CODE REVIEW

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java [hunk 1]
```diff
@@ -153,7 +155,8 @@ public PodTemplate(PodTemplate from) {
         this.setActiveDeadlineSeconds(from.getActiveDeadlineSeconds());
         this.setVolumes(from.getVolumes());
         this.setWorkspaceVolume(from.getWorkspaceVolume());
-        this.setYamls(from.getYamls());
+        this.setYaml(from.getYaml());
```
Comment (carlossg) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java]: bad merge ?

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java [hunk 2]
```diff
@@ -740,10 +752,12 @@ private String getContainersDescriptionForLogging() {
             }
             sb.append("\n");
         }
-        for (String yaml : getYamls()) {
-            sb.append("yaml:\n")
-                .append(yaml)
-                .append("\n");
+        if (getYamls().size() > 0 && isShowRawYaml()) {
```
Comment (carlossg) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java]: no need to check for `getYamls().size()`

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java (hunk 1 above)
Comment (joyarzun) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java]: Yep, my fault. Done

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java (hunk 2 above)
Comment (joyarzun) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java]: Done

**Comment by carlossg** [comment]
I can't push to the PR, can you add this commit ?
https://github.com/jenkinsci/kubernetes-plugin/commit/435e2fcda80fa95cd64db824295126ff324b68f0

**Comment by joyarzun** [comment]
> I can't push to the PR, can you add this commit ?
> [435e2fc](https://github.com/jenkinsci/kubernetes-plugin/commit/435e2fcda80fa95cd64db824295126ff324b68f0)

I did the change requested but why change boolean to Boolean?

**Comment by carlossg** [comment]
It's a tricky thing with Jenkins. On upgrade only the readResolve method is called, not the initializers. We need to set the value to true, so using a boolean we can see that if it is null it means we are upgrading and then set it to true.

**Comment by joyarzun** [comment]
> It's a tricky thing with Jenkins. On upgrade only the readResolve method is called, not the initializers. We need to set the value to true, so using a boolean we can see that if it is null it means we are upgrading and then set it to true.

Thanks for the explanation

**Comment by fredg02** [comment]
Thanks @joyarzun & @carlossg for picking this up and completing my PR. :+1:

**Pull request merged**